src/strategy_manager/strategy.manager.tests.cpp

```diff
@@ -32,9 +35,29 @@ namespace antara::mmbot::tests
         CHECK_NE(a1, a3);
     }
 
+    TEST_CASE ("market_making_strategy equality")
+    {
+        auto pair = antara::pair::of("A", "B");
+        auto spread = st_spread{0.1};
+        auto side = antara::side::sell;
+
+        auto q_10 = st_quantity{10};
+        auto q_11 = st_quantity{11};
+
+        auto mms_1 = market_making_strategy{pair, spread, q_10, side};
+        auto mms_2 = market_making_strategy{pair, spread, q_11, side};
+
+        CHECK_EQ(mms_1, mms_1);
+        CHECK_NE(mms_1, mms_2);
+    }
+
     TEST_CASE ("strats can be added and retreived")
     {
-        auto sm = strategy_manager();
+        dex dex;
+        cex cex;
+        order_manager_mock om(dex, cex);
+        auto ps = price_service_platform_mock();
+        auto sm = strategy_manager(ps, om);
```

Comment:
strategy_manager can be templated on the implementation class such as:

`strategy_manager<price_service_platform_fake>(ps, om);`

Reply:
Abstract classes and interfaces are useful when you have to manipulate several, not when you have to mock a class. The templates will do the trick here